packages/edit-post/src/components/editor-regions/index.js

```diff
@@ -8,8 +8,73 @@ import classnames from 'classnames';
  */
 import { navigateRegions } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useSimulatedMediaQuery } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 
 function EditorRegions( { footer, header, sidebar, content, publish, className } ) {
+	const resizableStylesheets = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSettings().resizableStylesheets;
+	}, [] );
+
+	const deviceType = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).deviceType();
+	} );
+
+	const [ actualWidth, updateActualWidth ] = useState( window.innerWidth );
+
+	window.addEventListener( 'resize', () => updateActualWidth( window.innerWidth ) );
+
+	const canvasWidth = ( device ) => {
+		let deviceWidth = 0;
+
+		switch ( device ) {
+			case 'Tablet':
+				deviceWidth = 780;
+				break;
+			case 'Mobile':
+				deviceWidth = 360;
+				break;
+			default:
+				deviceWidth = 2000;
+		}
+
+		return deviceWidth < actualWidth ? deviceWidth : actualWidth;
+	};
+
+	const marginValue = () => window.innerHeight < 800 ? 36 : 72;
+
+	const minHeightValue = ( device ) => device === 'Mobile' ? 640 : 768;
+
+	const contentInlineStyles = ( device ) => {
+		switch ( device ) {
+			case 'Tablet':
+			case 'Mobile':
+				return {
+					width: canvasWidth( device ),
+					margin: marginValue() + 'px auto',
+					flexGrow: 0,
+					minHeight: minHeightValue( device ) + 'px',
+				};
+			default:
+				return null;
+		}
+	};
+
+	const sidebarInlineStyles = ( device ) => {
+		switch ( device ) {
+			case 'Tablet':
+			case 'Mobile':
+				return {
+					minHeight: minHeightValue( device ) + ( marginValue() * 2 ) + 'px',
+				};
+			default:
+				return null;
+		}
+	};
```

Comment:
Maybe we could move all of the above code into a hook that returns `[ contentInlineStyles, sidebarInlineStyles ]` so that `EditorRegions` isn't too caught up with the specifics of how editor resizing works?